presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -1089,6 +1091,11 @@ public static Object invoke(ConnectorSession session, ScalarFunctionImplementati
         if (handle.type().parameterCount() > 0 && handle.type().parameterType(0) == ConnectorSession.class) {
             handle = handle.bindTo(session);
         }
+        Optional<Slice> returnValue = Optional.empty();
+        if (function.isReturnValueAsParameter()) {
+            Preconditions.checkArgument(handle.type().parameterType(0) == Slice.class, "only Slice supported as return parameter");
```

Comment:
It is supported. Note that above we bind session as first parameter and overwrite `handle`.
And then we operate on new `handle` with session already bound.